import "rec/gui/Color.proto";
import "rec/gui/Font.proto";
import "rec/widget/Widget.proto";

package rec.widget;

message ThreadDescription {
  optional uint32 priority = 1 [default = 6];
  optional uint32 period = 2   [default = 40];
};

message AudioThumbnailDesc {
  optional Widget widget = 1;
  optional uint32 cursor_thickness = 2                    [default = 1];
  optional uint32 source_samples_per_thumbnail_sample = 4 [default = 512];
  optional uint32 thumbnail_cache = 5                     [default = 5];
  optional ThreadDescription cursor_thread = 6;
  optional uint32 tic_height = 8                          [default = 8];
  optional uint32 subtic_height = 9                       [default = 3];
  optional bool use_captions = 10                         [default = true];
  optional bool tics_at_bottom = 11                       [default = true];
};
